src/core/util/SubGridSet.ts

```diff
@@ -0,0 +1,42 @@
+import { SubGrid } from "./SubGrid";
+import { TilePos } from "./TilePos";
+
+// A union of rectangular sections of grid
+export default class SubGridSet implements Iterable<TilePos> {
+  data : SubGrid[] = [];
+
+  add(subGrid: SubGrid): void {
+    this.data.push(subGrid);
+  }
+
+  has(cell: TilePos): boolean {
+    return this.data.some(subGrid => subGrid.has(cell));
+  }
+
+  [Symbol.iterator](): Iterator<TilePos> {
+    return this.values();
+  }
+
+  * cellGenerator(): Generator<TilePos> {
+    for (let i = 0; i < this.data.length; i++) {
+      const subGrid = this.data[i];
+      for (let x = subGrid.x; x < subGrid.x + subGrid.width; x++) {
+        for (let y = subGrid.y; y < subGrid.y + subGrid.height; y++) {
```

Comment:
I could make subgrid iterable and remove the x/y for loops.

Reply:
You could even use the `yield*` keyword then!
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/yield*

I just learned about it today.